evalml/tests/automl_tests/test_iterative_algorithm.py

```diff
@@ -303,11 +328,7 @@ def test_iterative_algorithm_pipeline_params(parameters, dummy_binary_pipeline_c
     # make sure that future batches remain in the hyperparam range
     for i in range(1, 5):
         next_batch = algo.next_batch()
-        for p in next_batch:
-            if isinstance(parameters, Categorical):
-                assert p.parameters['Mock Classifier']['dummy_parameter'] in parameters
-            else:
-                assert p.parameters['Mock Classifier']['dummy_parameter'] == parameter
+        assert all([p.parameters['Mock Classifier']['dummy_parameter'] in hyperparameters for p in next_batch])
```

Comment:
might be nice to add a sanity check that the algorithm is producing different parameters (first example it goes thrugh 1, 3, and 5 as well).

Reply:
Great idea